File: mns-reactor/src/test/java/cz/fb/manaus/reactor/betting/validator/ValidationMetricsCollectorTest.java
package cz.fb.manaus.reactor.betting.validator;

import com.codahale.metrics.MetricRegistry;
import cz.fb.manaus.core.model.Side;
import cz.fb.manaus.core.test.AbstractLocalTestCase;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.stream.Collectors;

import static org.junit.Assert.assertEquals;

public class ValidationMetricsCollectorTest extends AbstractLocalTestCase {

    public static final Validator VALIDATOR = context -> ValidationResult.ACCEPT;
    @Autowired
    private ValidationMetricsCollector metricsCollector;
    @Autowired
    private MetricRegistry metricRegistry;

    @Test
    public void testMetrics() {
        metricsCollector.updateMetrics(ValidationResult.ACCEPT, Side.BACK, VALIDATOR.getName());
        metricsCollector.updateMetrics(ValidationResult.REJECT, Side.BACK, VALIDATOR.getName());
        var keys = metricRegistry.getCounters().keySet().stream()
                .filter(key -> key.startsWith(ValidationMetricsCollector.PREFIX))
                .filter(key -> key.contains(VALIDATOR.getName()))
                .collect(Collectors.toList());

        assertEquals(2, keys.size());
        keys.forEach(key -> assertEquals(1L, metricRegistry.counter(key).getCount()));
    }
}

File: mns-reactor/src/test/java/cz/fb/manaus/reactor/betting/validator/ValidationMetricsCollectorTest.kt
package cz.fb.manaus.reactor.betting.validator

import cz.fb.manaus.core.makeName
import cz.fb.manaus.core.model.Side
import cz.fb.manaus.reactor.betting.BetEvent
import io.micrometer.core.instrument.Metrics
import org.junit.jupiter.api.Test
import kotlin.test.assertTrue

class TestValidator : Validator {
    override fun validate(event: BetEvent): ValidationResult {
        return ValidationResult.OK
    }
}

class ValidationMetricsCollectorTest {

    private val collector = ValidationMetricsCollector

    @Test
    fun `validation metrics`() {
        val validator = TestValidator()
        collector.updateMetrics(ValidationResult.OK, Side.BACK, makeName(validator))
        collector.updateMetrics(ValidationResult.DROP, Side.BACK, makeName(validator))
        val meters = Metrics.globalRegistry.meters
            .filter { it.id.name.startsWith("mns_validator_stats") }
        assertTrue { meters.size >= 2 }
    }

}